src/TypeScript/Analyzer/AnalyzerBase.cs

```diff
@@ -0,0 +1,109 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.Core.Telemetry;
+using SonarLint.VisualStudio.TypeScript.EslintBridgeClient;
+using SonarLint.VisualStudio.TypeScript.Rules;
+
+namespace SonarLint.VisualStudio.TypeScript.Analyzer
+{
+    internal abstract class AnalyzerBase : IDisposable
+    {
+        private readonly IThreadHandling threadHandling;
+
+        protected readonly ITelemetryManager telemetryManager;
+        protected readonly IAnalysisStatusNotifierFactory analysisStatusNotifierFactory;
+        protected readonly IEslintBridgeAnalyzer eslintBridgeAnalyzer;
+        protected IAnalysisStatusNotifier analysisStatusNotifier;
+
+        protected AnalyzerBase(ITelemetryManager telemetryManager,
+            IAnalysisStatusNotifierFactory analysisStatusNotifierFactory,
+            IEslintBridgeAnalyzerFactory eslintBridgeAnalyzerFactory,
+            IRulesProviderFactory rulesProviderFactory,
+            IEslintBridgeClient eslintBridgeClient,
+            IThreadHandling threadHandling,
+            string repoKey,
+            Language language)
+        {
+            this.threadHandling = threadHandling;
+
+            this.telemetryManager = telemetryManager;
+            this.analysisStatusNotifierFactory = analysisStatusNotifierFactory;
+
+            var rulesProvider = rulesProviderFactory.Create(repoKey, language);
+            eslintBridgeAnalyzer = eslintBridgeAnalyzerFactory.Create(rulesProvider, eslintBridgeClient);
+        }
+
+        protected async virtual Task<string> GetTsConfig(string sourceFilePath, CancellationToken cancellationToken)
+        {
+            return null;
+        }
+
+        internal async Task ExecuteAsync(string languageKey, string analyzerName, string filePath, IIssueConsumer consumer, CancellationToken cancellationToken)
+        {
+            telemetryManager.LanguageAnalyzed(languageKey);
+            analysisStatusNotifier = analysisStatusNotifierFactory.Create(analyzerName, filePath);
+
+            await threadHandling.SwitchToBackgroundThread();
+            analysisStatusNotifier.AnalysisStarted();
+
+            try
+            {
+                var tsConfig = await GetTsConfig(filePath, cancellationToken);
+
+                if (tsConfig == null && languageKey == "ts") { return; }
```

Comment:
Task<(bool isError, tsconfig)> TryGetTsConfig
return when isError == true
in other cases, continue

I don't like that we use telemetry language key for logic

Reply:
The default implementation would always return (false, null)

Reply:
Or some other way of indicating that's it in one of the 3 states ("not required", "all good" and "required, but missing")